rpm/0007-Add-RFC7636-aka-PKCE-support.patch

```diff
@@ -110,7 +111,7 @@ index a5f8e9769939ad1e0fe04840ef9208365167369b..452d03a88e927a26a5860a572efe98ef
          url.addQueryItem(SCOPE, QUrl::toPercentEncoding(scopes.join(" ")));
      }
 +    // PKCE (RFC7636)
-+    if (!d->m_oauth2Data.SkipPKCE()) {
++    if (!d->m_oauth2Data.SkipPKCE() &&!responseType.contains("token")) {
```

Comment:
I think this should be 
```C++
if (!d->m_oauth2Data.SkipPKCE() && responseType.contains("code")) {
```
since [RFC 7636 section 4.3](https://www.rfc-editor.org/rfc/rfc7636#section-4.3) defines that the code is sent with "Authorization Request" as the only situation when it would be sent.

I guess if some extensions want to use it, then they would need changes of course but that  support needs to be added on top of the RFCs anyway.

Also pondering a bit if `"code"` should be a constant at the beginning of the file, but that's not that important. There is already `AUTH_CODE` there but that's for a different purpose and I'm not sure how I'd feel about (ab)using it.

Reply:
>  Also pondering a bit if "code" should be a constant at the beginning of the file, but that's not that important. 
> There is already AUTH_CODE there but that's for a different purpose and I'm not sure how I'd feel about (ab)using it.

I checked the AUTH_CODE constant it is already used in similar use cases, however I'm not sure
if it makes sense to do such refactoring alone. 
I would ask upstream when posting these patches.

Reply:
Please don't get stuck with that `"code"`, it's fine as a string or reusing that `AUTH_CODE`. I will approve it either way, that first part of my comment is much more important. :slightly_smiling_face: